packages/main/src/plugin/kubernetes/kubernetes-client.ts

```diff
@@ -365,6 +365,72 @@ export class KubernetesClient {
     return kubeContexts;
   }
 
+  async duplicateContext(contextName: string): Promise<Context[]> {
```

Comment:
This is only one big function with side effects (the contexts will be changed in place). 

I think it would be more testable and maintainable if we have 'pure' (without any side effect) functions as most as possible, for example a functio. which returns the new context, then another which adds this context to a list of contexts, then a last ones which updates the current contexts.